x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/IndexLifecycleRunner.java

```diff
@@ -154,6 +156,48 @@ public void onFailure(Exception e) {
         }
     }
 
+    private void onErrorMaybeRetryFailedStep(String policy, IndexMetaData indexMetaData) {
+        String index = indexMetaData.getIndex().getName();
+        LifecycleExecutionState lifecycleState = LifecycleExecutionState.fromIndexMetadata(indexMetaData);
+        Step failedStep = stepRegistry.getStep(indexMetaData, new StepKey(lifecycleState.getPhase(), lifecycleState.getAction(),
+            lifecycleState.getFailedStep()));
+        if (failedStep == null) {
+            logger.warn("failed step [{}] is not part of policy [{}] anymore, or it is invalid. skipping execution",
+                lifecycleState.getFailedStep(), policy);
+            return;
+        }
+
+        if (failedStep.isRetryable()) {
+            if (lifecycleState.isTransitiveError()) {
+                int retryCount = lifecycleState.getFailedStepRetryCount() == null ? 0 : lifecycleState.getFailedStepRetryCount();
+                Integer maxRetriesCount =
+                    indexMetaData.getSettings().getAsInt(LifecycleSettings.LIFECYCLE_MAX_FAILED_STEP_RETRIES_COUNT, 15);
```

Comment:
This shouldn't specify a default since a default is already specified in the setting itself (otherwise we have to change two places if it's changed in the future)